src/Native/gc/env/gcenv.interlocked.inl

```diff
@@ -193,7 +193,7 @@ __forceinline T Interlocked::CompareExchangePointer(T volatile *destination, T e
     return (T)(TADDR)_InterlockedCompareExchange((long volatile *)(void* volatile *)destination, (long)(void*)exchange, (long)(void*)comparand);
 #endif
 #else
-    return (T)(TADDR)__sync_val_compare_and_swap((void* volatile *)destination, comparand, exchange);
+    return (T)(TADDR)__sync_val_compare_and_swap((void* volatile *)destination, comparand, reinterpret_cast<void*>(exchange));
```

Comment:
For consistency, could you please add the cast to `comparand` as well, and use C-style casts since they are used everywhere else in this file?